### ISSUE TITLE
[BUG][Fuzzer][wrong-code] Whole-tile `T.copy` of an out-of-bounds tile silently ignores `T.annotate_safe_value` and zero-fills the out-of-bounds region instead of using the annotated value

### ISSUE THREAD

**Issue by Hughshine** [tile-ai/tilelang#2543]
### Required prerequisites

- [x] I have read the documentation <https://tilelang.com>.
- [x] I have searched the [Issue Tracker](https://github.com/tile-ai/tilelang/issues) that this hasn't already been reported. (comment there if it has.)

### What version of TileLang are you using?

0.1.11

### System information

NVIDIA H100 (sm_90), CUDA 13.0, from-source `v0.1.11` (`cd37ed5fc35ae7a60a1277c8eb49028174ac51e6`). The defect is in the tile-copy SIMT lowering (`src/op/copy.cc`), which is target-independent; reproduced on sm_90.

### Problem description

A whole-tile `T.copy` of an out-of-bounds global tile into shared memory silently ignores `T.annotate_safe_value` and always zero-fills the out-of-bounds region. Copying a `(60, 60)` `int32` tensor into a `(64, 64)` tile annotated with safe value `-99` produces `0` (not `-99`) in the out-of-bounds rows/cols `[60..63]`; the string `-99` never appears in the generated CUDA. The in-bounds region `[0:60, 0:60]` is correct, the kernel compiles and runs with no error, and the wrong fill is deterministic. Any code relying on the padding (e.g. a masked softmax over a padded tile) then consumes `0` where it asked for `-99`.

The value *is* a real, honored feature on the elementwise path: a byte-identical per-element `T.Parallel` load of the same tile with the same annotation returns `-99`. Only the whole-tile `T.copy` lowering drops it.

### Reproducible example code

```python
import tilelang
tilelang.disable_cache()
import tilelang.language as T
import torch

M = N = 60      # global 60x60
BM = BN = 64    # tile 64x64 -> rows/cols [60..63] are out of bounds


@T.prim_func
def bulk(A: T.Tensor((M, N), "int32"), Out: T.Tensor((BM, BN), "int32")):
    with T.Kernel(1, threads=128) as _:
        As = T.alloc_shared((BM, BN), "int32")
        T.annotate_safe_value({A: -99})          # request -99 as the OOB fill
        T.copy(A[0:BM, 0:BN], As)                # whole-tile copy
        for i, j in T.Parallel(BM, BN):
            Out[i, j] = As[i, j]


k = tilelang.compile(bulk, out_idx=[1], target="cuda")
A = torch.arange(M * N, dtype=torch.int32, device="cuda").reshape(M, N)
Out = k(A)
print("'-99' in CUDA:", "-99" in k.get_kernel_source())   # False
print("OOB row 60 [0:8]:", Out[60, :8].cpu().tolist())     # [0,0,0,0,0,0,0,0]  (want -99)
print("inbounds correct:", torch.equal(Out[:60, :60].cpu(), A.cpu()))  # True
```

Replacing the `T.copy` with a per-element `for i, j in T.Parallel: As[i,j] = A[i,j]` load — same annotation — instead prints `'-99' in CUDA: True` and `[-99, -99, …]`, which is the control isolating the defect to the whole-tile lowering.

### Traceback

```pytb
No traceback — the kernel compiles and runs to completion; the OOB fill is silently `0` instead of the annotated `-99`.
```

### Expected behavior

A whole-tile `T.copy` of an OOB tile should fill the out-of-bounds region with the value passed to `T.annotate_safe_value` (produce `-99`), matching the elementwise load of the same tile with the same annotation. The two paths honoring the same annotation differently is an internal inconsistency, not a design choice.

### Additional context

**Root cause.** The OOB fill of a tile copy is emitted by `CopyNode::MakeSIMTLoop` ([`src/op/copy.cc#L452`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/op/copy.cc#L452)), which builds the guarded load with a hardcoded zero on the false branch ([`#L485`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/op/copy.cc#L485)):

```cpp
value = if_then_else(src_predicate, value, make_zero(dst->dtype));
```

This runs inside `LowerTileOp`, *before* `LegalizeSafeMemoryAccess`. So the load arrives at the legalization pass already guarded with `make_zero`, and the pass's safe-value machinery — which reads `attr::kSafeValueMap` and applies `GetSafeValue` to still-unguarded loads ([`legalize_safe_memory_access.cc#L631`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/transform/legalize_safe_memory_access.cc#L631), [`#L356`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/transform/legalize_safe_memory_access.cc#L356)) — never sees this fill and never consults the annotated value. The elementwise path stays unguarded through `LowerTileOp`, so the legalization pass reaches it and honors the annotation — exactly the asymmetry the control shows.

**Suggested fix.** Thread the annotated safe value into `MakeSIMTLoop`'s false branch instead of hardcoding `make_zero`: look up the source buffer's safe value (defaulting to `make_zero`, mirroring `GetSafeValue`) where the copy is lowered, and use it at [`copy.cc#L485`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/op/copy.cc#L485). This requires making `attr::kSafeValueMap` available to the copy lowering (it is on the block `LowerTileOp` processes). A local patch of this shape flips the repro to `-99` in the OOB region with the in-bounds region and the no-annotation zero-fill default both unchanged.

**Provenance.** The safe-value mechanism was added to `LegalizeSafeMemoryAccess` in [`72111642a`](https://github.com/tile-ai/tilelang/commit/72111642a) (#1050, 2025-10-17); the whole-tile copy path's `make_zero` fill in `MakeSIMTLoop` predates and was never wired to it.

**Not a duplicate.** Distinct from the TMA-copy *crash* reports (`cudaErrorIllegalInstruction` / launch abort / compile abort from TMA alignment/stride/swizzle constraints) — this is a silent wrong result on a clean-compiling kernel. The wrong fill comes from the SIMT copy lowering's `make_zero`, not from the TMA descriptor's `oob_fill` (flipping that descriptor field is inert for this OOB copy, since an OOB tile suppresses bulk/TMA selection). No open or closed tracker issue covers the whole-tile copy dropping `annotate_safe_value`.
